frontend/apps/reader/modules/readergesture.lua

```diff
@@ -1577,11 +1577,6 @@ function ReaderGesture:onToggleReadingOrder()
     local document_module = self.ui.document.info.has_pages and self.ui.paging or self.ui.rolling
     document_module.inverse_reading_order = not document_module.inverse_reading_order
     document_module:setupTouchZones()
-    -- Needed to reset the touch zone overrides
```

Comment:
So this part was the ugly hack?

Reply:
Well, just _a_ hack :) I don't think we ever did remove & re-add before, so that little bug in that trusted but forgotten bit of code was staying unnoticed.